### ISSUE TITLE
Omniauth handling all my exceptions

### ISSUE THREAD

**Issue by robinmonjo** [omniauth/omniauth#1049]
Please complete all sections.

### Configuration

- Provider Gem: `omniauth-saml`, `omniauth-facebook`, `omniauth-linkedin`
- Ruby Version: `2.6.8`
- Framework: `Rails`
- Platform: `Linux/Mac`

### Expected Behavior

Omniauth + Rails with multiple providers. When I have an exception in my code that has nothing to do with omniauth, the omniauth middleware should do nothing. 

### Actual Behavior

The omniauth middleware handle the exception

### Steps to Reproduce

In this piece of code (strategy.rb):

```
    def call!(env)
      unless env['rack.session']
        error = OmniAuth::NoSessionError.new('You must provide a session to use OmniAuth.')
        raise(error)
      end

      warn_if_using_get

      @env = env
      @env['omniauth.strategy'] = self if on_auth_path?

      return mock_call!(env) if OmniAuth.config.test_mode

      begin
        return options_call if on_auth_path? && options_request?
        return request_call if on_request_path? && OmniAuth.config.allowed_request_methods.include?(request.request_method.downcase.to_sym)
        return callback_call if on_callback_path?
        return other_phase if respond_to?(:other_phase)
      rescue StandardError => e
        return fail!(e.message, e)
      end

      @app.call(env)
    end
```
If my code is wrong for some reasons and create an exception this get rescued and the `fail!` method is called. But, **@env['omniauth.strategy']** is nil because I'm not at all on a path that should require any omniauth logic.
I load multiple providers looks like the strategy that is concerned here is the last one loaded (saml), and so:

```
  def fail!(message_key, exception = nil)
      env['omniauth.error'] = exception
      env['omniauth.error.type'] = message_key.to_sym
      env['omniauth.error.strategy'] = self

      if exception
        log :error, "Authentication failure! #{message_key}: #{exception.class}, #{exception.message}"
      else
        log :error, "Authentication failure! #{message_key} encountered."
      end

      OmniAuth.config.on_failure.call(env)
    end
```
at the end of the middleware I got:

- env['omniauth.error'] ➡️ my exception
- env['omniauth.error.strategy'] ➡️ saml
- env['omniauth.strategy'] ➡️ nil

I would expect that when no strategy is found, the middleware does nothing and just forward the request to the next one. Something like:

```
@env['omniauth.strategy'] = self if on_auth_path?
return @app.call(env) if @env['omniauth.strategy'].nil? # new code, no strategy should handle this request
```

**Comment by robinmonjo** [comment]
Looks like this was handled in this commit: https://github.com/omniauth/omniauth/commit/6f4cdb08f4741a0b3c83dfafd5f93e972be97c0e I'll update and see how it goes

**Comment by robinmonjo** [comment]
Confirm this was fixed upgrading from 2.0.0 to 2.0.4 solved my issue, sorry for the noise
